sqlglot/lineage.py

```diff
@@ -276,8 +276,46 @@ def to_node(
                 reference_node_name=reference_node_name,
                 trim_selects=trim_selects,
             )
+
+        elif scope.pivots:
+            pivot = next(
+                (p for p in scope.pivots if p.alias_or_name == c.table and not p.unpivot), None
+            )
+            if pivot:
+                # The source is a pivot operation, so we need to trace back to the aggregated column
+                pivot_column_mapping = {}
+                pivot_aggs = len(pivot.expressions)
+                columns = pivot.args.get("columns")
+
+                for i, agg in enumerate(pivot.expressions):
+                    agg_cols = list(agg.find_all(exp.Column))
+                    for col in range(i, len(columns), pivot_aggs):
+                        # Go through the field_names in a step manner for this aggfunc
+                        pivot_column_mapping[columns[col].name] = agg_cols
+
+                for agg_column in pivot_column_mapping[c.alias_or_name]:
+                    table = agg_column.table
+                    source = scope.sources.get(table)
+                    if isinstance(source, Scope):
+                        to_node(
+                            agg_column.name,
+                            scope=source,
+                            scope_name=table,
+                            dialect=dialect,
+                            upstream=node,
+                            trim_selects=trim_selects,
+                        )
```

Comment:
I think I am missing here the `source_name` and `reference_node_name` fields.

Is it right that `reference_node_name` references a CTE name and `source_name` the table names?

If so what are the right values for a pivot operation here?

Reply:
@VaggelisD can you help me out here please?

Reply:
The `source_name` is the name of a source table that got expanded into an aliased derived table (subquery):

```sql
-- The original source name is t2
SELECT x FROM t2

-- After expanding t2, it's an aliased derived table; we can map _q_0 to t2, the source_name
SELECT x FROM (SELECT y FROM t1) _q_0
```

The `reference_node_name` is for the name of the CTE or the alias of the derived table a node is from.

If I'm not mistaken, we want to mimic what we do in the previous branch, but for the aggregated column?

```python
                source_name=source_names.get(table) or source_name,
                reference_node_name=reference_node_name,
```

Reply:
I added the the parameters and added asserts in tests